tests/test_scheduler.py

```diff
@@ -2190,3 +2176,61 @@ def test_scheduler_unqualified_imports(config):
 
     assert item.enable_imports
     assert item.children == ('other_routine',)
+
+
+def test_transformation_config(config):
+    """
+    Test the correct instantiation of :any:`Transformation` objecst from config
+    """
+    my_config = config.copy()
+    my_config['transformations'] = {
+        'DependencyTransformation': {
+            'module': 'loki.transform',
+            'options':
+            {
+                'suffix': '_rick',
+                'module_suffix': '_roll',
+                'replace_ignore_items': False,
+            }
+        }
+    }
+    cfg = SchedulerConfig.from_dict(my_config)
+    assert cfg.transformations['DependencyTransformation']
+
+    transformation = cfg.transformations['DependencyTransformation']
+    assert isinstance(transformation, DependencyTransformation)
+    assert transformation.suffix == '_rick'
+    assert transformation.module_suffix == '_roll'
+    assert not transformation.replace_ignore_items
+
+
+def test_transformation_config_with_dimension(config):
+    """
+    Test instantiation of :any:`Transformation` from config with
+    :any:`Dimension` argument.
+    """
+    from transformations.single_column_coalesced import SCCBaseTransformation  # pylint: disable=import-outside-toplevel
```

Comment:
Not sure I'm a fan of using a transformation from the "transformations" package within the core Loki testbase - This introduces a circular dependency between the two packages.
Could we not write a dummy transformation to file on the fly and load that instead?

Reply:
Agreed. Added a new dummy test to pick this up properly, which incidentally now also covers the custom import-path case. :smile: